AerospikeClient/Async/AsyncCommand.cs

```diff
@@ -229,29 +241,60 @@ private void ExecuteCommand()
 			}
 			catch (AerospikeException.Connection aec)
 			{
+				exception = aec;
 				ErrorCount++;
 				ConnectionFailed(aec);
 			}
 			catch (AerospikeException.Backoff aeb)
 			{
+				exception = aeb;
 				ErrorCount++;
 				Backoff(aeb);
 			}
 			catch (AerospikeException ae)
 			{
+				exception = ae;
 				ErrorCount++;
 				FailOnApplicationError(ae);
 			}
 			catch (SocketException se)
 			{
+				exception = se;
 				ErrorCount++;
 				OnSocketError(se.SocketErrorCode);
 			}
 			catch (Exception e)
 			{
+				exception = e;
 				ErrorCount++;
 				FailOnApplicationError(new AerospikeException(e));
 			}
+			finally
+			{
+				if (activity != null && exception != null)
+				{
+					activity.SetStatus(ActivityStatusCode.Error, exception.ToString());
+				}
+			}
+		}
+
+		private void StartActivity()
+		{
+			// Because of the retry mecanism, one operation can result in several I/Os, on several nodes.
+			// For this reason, we create a new Activity on every retry, so that we can set the actual node
+			// adress and the I/O status as tags of this current Activity.
+
+			// If we're retrying, stop the previous Activity before restarting a new one.
+			activity?.Dispose();
+			activity = Tracing.Source.StartActivity(GetType().Name, ActivityKind.Client);
+			if (activity != null)
+			{
+				// https://github.com/open-telemetry/opentelemetry-specification/blob/6ce62202e5407518e19c56c445c13682ef51a51d/specification/trace/semantic_conventions/span-general.md#general-remote-service-attributes
+				activity.SetTag("peer.service", Tracing.SERVICE_NAME);
```

Comment:
The service name is one of the most important information. When the trace is displayed on a UI (Jaeger, Tempo), it will be the most visible info. I would avoid hardcoding it but either use the cluster name or a new user-provided string.